### PR TITLE
Add clear search functionality to Search component

### PR DESCRIPTION
**Pull request by duhnnie**
Adds the clear-search button to Search component

### CODE REVIEW

File: src/components/search/search.js [hunk 1]
```diff
@@ -1,26 +1,41 @@
...
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      searchValue: ''
+    };
+  }
+
+  clear () {
+    this.setState({searchValue: ''});;
```
Comment (ougarte) [inline src/components/search/search.js]: I think, you need only one semi-colon

File: src/components/search/search.js [hunk 2]
```diff
@@ -1,26 +1,41 @@
...
-            placeholder={this.props.placeholder}
-            size='80'
-            onKeyDown={(e, input) => this.handleKeyPress(e, input)}/>
-          <i className='search icon'></i>
-        </div>
-      </div>
+      <Input
+        value = {this.state.searchValue}
+        fluid
+        onKeyDown={ (e, input) => this.handleKeyPress(e, input) }
```
Comment (ougarte) [inline src/components/search/search.js]: I recommend to use `this.handleKeyPress = this.handleKeyPress.bind(this);` on the constructor

Comment (Jhegs) [reply src/components/search/search.js]: bind() should die 😅 long live to arrow funcitons

Comment (ougarte) [reply src/components/search/search.js]: please don't let bind to die ... it was (and still is) my friend in bad moments  😅

Comment (duhnnie) [reply src/components/search/search.js]: with arrow functions we can't make use of `arguments` or call them in other contexts.

Comment (duhnnie) [reply src/components/search/search.js]: So, which path should I follow? ougarte's or Jhegs?

Comment (duhnnie) [reply src/components/search/search.js]: I think I'm gonna go the ougarte recommendation.

File: src/components/search/search.js (hunk 1 above)
Comment (duhnnie) [reply src/components/search/search.js]: fixed

**Comment by duhnnie** [comment]
PR was updated with received suggestions.

**Comment by duhnnie** [comment]
The PR was updated respect the **develop** branch (rebased).

File: src/components/search/search.js [hunk 3]
```diff
@@ -1,26 +1,43 @@
...
-      </div>
+      <Input
+        value = {this.state.searchValue}
+        fluid
+        onKeyDown={ this.handleKeyPress }
+        onChange = { e => this.setState({searchValue: e.currentTarget.value}) }
+        icon='search'
+        iconPosition='left'
+        placeholder={this.props.placeholder}
+        action={<Button icon='delete' onClick={() => this.clear() } />}
```
Comment (veymarjoel) [inline src/components/search/search.js]: What about put the button component in another line, to better understand the code. any suggestion @ougarte @JCFlow @Jhegs

Comment (duhnnie) [reply src/components/search/search.js]: done

**Comment by duhnnie** [comment]
PR was rebased against develop branch.

**Pull request merged**